File: stock-ingestor/internal/domain/entities/stock.go
package entities

import "time"

type StockData struct {
	Date           time.Time `json:"date"`
	Open           float64   `json:"open"`
	High           float64   `json:"high"`
	Low            float64   `json:"low"`
	Close          float64   `json:"close"`
	AdjustedClose  float64   `json:"adjusted_close"`
	Volume         int64     `json:"volume"`
	DividendAmount float64   `json:"dividend_amount"`
}

type StockTimeSeries struct {
	Symbol string               `json:"symbol"`
	Data   map[string]StockData `json:"data"`
}

File: frontend/src/lib/types/stock.ts
export interface StockData {
	date: Date;
	open: number;
	high: number;
	low: number;
	close: number;
	volume?: number;
}

export interface StockResponse {
	symbol: string;
	data: StockData[];
	metadata: {
		currency: string;
		interval: string;
		lastRefreshed: Date;
		timeZone: string;
	};
	overview?: CompanyOverview;
}

export interface CompanyOverview {
	symbol?: string;
	name?: string;
	exchange?: string;
	sector?: string;
	industry?: string;
	description?: string;
	country?: string;
	currency?: string;
	market_cap?: number;
	pe_ratio?: number;
	forward_pe_ratio?: number;
	eps?: number;
	dividend_per_share?: number;
	dividend_yield?: number;
	beta?: number;
	revenue_ttm?: number;
	profit_margin?: number;
	roa_ttm?: number;
	roe_ttm?: number;
	analyst_target_price?: number;
	week_52_high?: number;
	week_52_low?: number;
	book_value?: number;
	price_to_book?: number;
	shares_outstanding?: number;
}

export interface ChartDimensions {
	width: number;
	height: number;
	margin: {
		top: number;
		right: number;
		bottom: number;
		left: number;
	};
}

export interface TooltipData {
	date: Date;
	open: number;
	high: number;
	low: number;
	close: number;
	volume?: number;
	x: number;
	y: number;
}
